Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/setup.py

```diff
@@ -198,6 +201,42 @@ def _fork_remote(cls, origin, username, name):
             return '{}:{}/{}.git'.format(origin.split(':')[0], username, name)
         return None
 
+    @classmethod
+    def _security_levels(cls, repository, username=None):
+        proc = run(
+            [local.Git.executable(), 'config', '--get-regexp', 'webkitscmpy.remotes'],
+            capture_output=True, cwd=repository.root_path,
+            encoding='utf-8',
+        )
+        if proc.returncode:
+            return {}
+
+        level = 0
+        result = {}
+        for line in proc.stdout.splitlines():
+            key, value = line.split(' ', 1)
+            match = cls.REMOTE_RE.match(value)
+            if match:
+                result['{}:{}'.format(match.group('host'), match.group('path'))] = level
+                if username:
+                    team, repository = match.group('path').split('/', 1)
+                    if team == repository:
+                        forked_name = repository
+                    elif '-' in repository:
+                        forked_name = repository
+                    else:
+                        forked_name = '{}-{}'.format(repository, team)
+                    result['{}:{}/{}'.format(match.group('host'), username, forked_name)] = level
+
+            result[str(key.split('.')[-1])] = level
+            if username:
+                if key.split('.')[-1] == 'origin':
+                    result['fork'] = level
+                else:
+                    result['{}-fork'.format(key.split('.')[-1])] = level
```

Comment:
IIUC, this is assumes the naming convention of forks that git-webkit sets up. And if a user has renamed their fork in GitHub or changed their remote name locally, it'll be caught by pre-push as having "no" security level.

Not leaking security content when someone renames their fork is obviously good, but I don't like putting more naming requirements on things which belong an engineer (namely, their repositories and local configuration). For example, I've mentioned to you before that I call my forks `emw`, because `emw-apple` is superfluous. It seems like I wouldn't be able to push security commits to a similarly-named private remote.

If we changed this, what would you think about:
- using the GitHub API to check whether a repo is private or not
- determining security level by identifying a known branch or tag on the remote

Reply:
@emw-apple discussed this offline, and checking using GitHub's API is a good idea, but also an expensive thing to do every push. `git-webkit setup` owning the API calls, though, makes a lot of sense. So that's what the most recent iteration of this PR does. That lets us rip out all logic that treats different remote names differently, everything is based on remote URL.